src/marqo/core/index_management/index_management.py

```diff
@@ -55,8 +55,27 @@ class IndexManagement:
         '''
     )
 
-    def __init__(self, vespa_client: VespaClient):
+    _DEPLOYMENT_LOCK_PATH = '/marqo__deployment_lock'
+    _DEPLOYMENT_LOCK_MAX_LOCK_PERIOD = 120
+    _DEPLOYMENT_LOCK_WATCHDOG_INTERVAL = 5
+
+    def __init__(self, vespa_client: VespaClient, zookeeper_client: Optional[MarqoKazooClient] = None):
         self.vespa_client = vespa_client
+        self.zookeeper_client = zookeeper_client
+        self.deployment_lock: Optional[DeploymentLock] = self._instantiate_deployment_lock()
+
+    def _instantiate_deployment_lock(self) -> Optional[DeploymentLock]:
+        """Instantiate a DeploymentLock"""
+        if self.zookeeper_client is None:
```

Comment:
I believe we discussed making zookeeper required and not allowing deployments without it?